File: android-app/app/src/main/java/com/runningtrainer/android/domain/contracts/PlanContracts.kt
package com.runningtrainer.android.domain.contracts

import com.runningtrainer.android.domain.model.FitnessLevel
import com.runningtrainer.android.domain.model.GoalType
import com.runningtrainer.android.domain.model.TrainingPlan
import java.time.LocalDate

data class PlanGenerationRequest(
    val goalType: GoalType,
    val fitnessLevel: FitnessLevel,
    val trainingDaysPerWeek: Int,
    val raceDate: LocalDate? = null,
    val durationWeeks: Int? = null,
    val startDate: LocalDate? = null,
    val age: Int? = null
)

data class PlanGenerationMetadata(
    val recoveryIntervalWeeks: Int,
    val progressionRate: Double,
    val taperApplied: Boolean
)

data class PlanGenerationResult(
    val plan: TrainingPlan,
    val metadata: PlanGenerationMetadata
)

File: ios-app/RunningTrainerNative/RunningTrainer/Domain/Contracts/PlanContracts.swift
import Foundation

struct PlanGenerationRequest {
    var goalType: GoalType
    var fitnessLevel: FitnessLevel
    var trainingDaysPerWeek: Int
    var raceDate: Date?
    var durationWeeks: Int?
    var startDate: Date?
    var age: Int?
}

struct PlanGenerationMetadata {
    var recoveryIntervalWeeks: Int
    var progressionRate: Double
    var taperApplied: Bool
}

struct PlanGenerationResult {
    var plan: TrainingPlan
    var metadata: PlanGenerationMetadata
}
